### PR TITLE
Add signals handler

### PR DESCRIPTION
**Pull request by kuznetsss**
For #442.

### CODE REVIEW

File: src/util/SignalsHandler.cpp [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        ASSERT(handler_ != nullptr, "SignalsHandler is not initialized");
+        handler_->secondSignalHandler_(signal);
+    }
+};
+
+SignalsHandler* SignalsHandlerStatic::handler_ = nullptr;
+
+}  // namespace impl
+
+SignalsHandler::SignalsHandler(Config const& config, std::function<void(std::string)> forceExitHandler)
```
Comment (cindyyan317) [inline src/util/SignalsHandler.cpp]: I am thinking the type of forceExitHandler std::function<void(std::string)> . Do we really need std::string here? It seems only for logging. How about log in lambda ?

File: src/util/SignalsHandler.cpp [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    })
+    , secondSignalHandler_([forceExitHandler = std::move(forceExitHandler)](int) {
+        forceExitHandler("Force exit on second signal.");
+        setHandler();
+    })
+{
+    impl::SignalsHandlerStatic::registerHandler(*this);
+
+    auto const gracefulPeriod =
+        std::round(config.valueOr("graceful_period", 10.f) * static_cast<float>(util::MILLISECONDS_PER_SECOND));
```
Comment (cindyyan317) [inline src/util/SignalsHandler.cpp:109]: Do we need to add this to docs and example config?

File: src/util/SignalsHandler.cpp (hunk 1 above)
Comment (kuznetsss) [reply src/util/SignalsHandler.cpp]: Good idea, thanks!

File: src/util/SignalsHandler.cpp (hunk 2 above)
Comment (kuznetsss) [reply src/util/SignalsHandler.cpp:109]: This PR doesn't implement graceful period feature, it is only a part of it. But I added docs and config example about it.

**Review by cindyyan317 — APPROVED** [review]
😆  Unittests have been fixed ~

**Review by godexsoft — APPROVED** [review]
Very nice 🚀

**Pull request merged**